mobile/native/src/ln_dlc/mod.rs

```diff
@@ -760,7 +760,40 @@ pub fn create_onboarding_invoice(amount_sats: u64, liquidity_option_id: i32) ->
     runtime.block_on(async {
         let node = NODE.get();
         let client = reqwest_client();
-        let user_channel_id = UserChannelId::new();
+
+        // check if we have already announced a channel before. If so we can reuse the `user_channel_id`
+        // instead of generating a new. This will reduce the garbarge record we create for everytime
+        // the user navigates to the invoice screen.
+        let channel = db::get_announced_channel(config::get_coordinator_info().pubkey)?;
+
+        let user_channel_id = match channel {
+            Some(channel) => channel.user_channel_id,
+            None => {
+                let user_channel_id = UserChannelId::new();
+                let channel = Channel::new(
+                    user_channel_id,
+                    0,
+                    0,
+                    config::get_coordinator_info().pubkey,
+                    liquidity_option_id,
+                );
```

Comment:
:thinking: I'm confused: why are we doing this twice (also below) now?

Reply:
that was some rebase error. thanks for spotting it.. I removed the unnecessary second time.